### PR TITLE
Replace int64 to uint64 for Trace and Span IDs

### PR DESCRIPTION
**Pull request by bg451**
This remove the possibility that an ID can overflow the max int64 value
and end up as negative.

Addresses https://github.com/opentracing/basictracer-go/issues/20

### CODE REVIEW

File: propagation_ot.go [hunk 1]
```diff
@@ -41,8 +41,8 @@ func (p *textMapPropagator) Inject(
 	if !ok {
 		return opentracing.ErrInvalidCarrier
 	}
-	carrier.Set(fieldNameTraceID, strconv.FormatInt(sc.raw.TraceID, 16))
-	carrier.Set(fieldNameSpanID, strconv.FormatInt(sc.raw.SpanID, 16))
+	carrier.Set(fieldNameTraceID, strconv.FormatUint(sc.raw.TraceID, 16))
+	carrier.Set(fieldNameSpanID, strconv.FormatUint(sc.raw.SpanID, 16))
```
Comment (yurishkuro) [inline propagation_ot.go]: no parent ID?

Comment (bg451) [inline propagation_ot.go]: the propagated span id ends up becoming the parent id for the new span.

https://github.com/opentracing/basictracer-go/blob/bg/make_id_uint64/propagation_ot.go#L112

**Comment by bensigelman** [comment]
LGTM. I ended up doing this in parallel last night (you beat me to sending it out) because we had unsigned formatted uint64s that were too large to represent as signed int64s...

**Comment by bg451** [comment]
Going to merge this in 24 hours if there aren't any objections.

**Pull request merged**